packages/core/src/modules/connections/ConnectionsModule.ts

```diff
@@ -212,26 +216,26 @@ export class ConnectionsModule {
     return this.connectionService.deleteById(connectionId)
   }
 
-  /**
-   * Find connection by verkey.
-   *
-   * @param verkey the verkey to search for
-   * @returns the connection record, or null if not found
-   * @throws {RecordDuplicateError} if multiple connections are found for the given verkey
-   */
-  public findByVerkey(verkey: string): Promise<ConnectionRecord | null> {
-    return this.connectionService.findByVerkey(verkey)
-  }
+  public async findByKeys({ senderKey, recipientKey }: { senderKey: string; recipientKey: string }) {
+    let connectionRecord
+    const theirDidRs = await this.didRepository.findMultipleByVerkey(senderKey)
```

Comment:
It's currently possible because we don't check if there is did doc for a given sender key. We should add this check. In the same way, it doesn't make sense to have more did doc for the same recipient key, does it?

My previous probably bad assumption was that I can create a connection request with a different did doc but signed by the same sender key as a previously created connection (for reusable invitation), then the creation of such did doc record won't fail on duplicated id error.